### PR TITLE
Enable the contract viewer

### PR DESCRIPTION
**Pull request by 2xic**
Some polish of the contract viewer
- Replace the emojis with real icons
- Make the 4byte text copy-able. 
- Various styling changes (re-use existing button component and styling)
- Some improved error handling 
- Added spinner for when the contract is being loaded
- Fixed the theme being wrong for the code editor on the initial load
- Fixed the updating of the URL not being changed on delete

Still some known issue that we will likely tackle later
- error handling: if you try to read from a mapping we can't decode it and it will give an error. 
- you can click on the input boxes of results node, but they aren't editable
- the encoded results could have been padded to better align (see photo - 4bytes are unaligned with the rest, we could have just padded the 4 bytes with empty space to make it all aligned)
Screenshot 2024-04-17 at 22 33 23

- Some minor styling changes here and there for the various sections inside the function interactions. 
- Last but not least, there are some improved that could be done to make the code more readable and reusable 

## Test plan
- Most of the changes here are styling, but do make sure that things work as before
     - Loading from [url](https://evm-codes-git-feature-enable-contract-viewer-smlxl.vercel.app/contract?address=0x00000000000000adc04c56bf30ac9d3c0aaf14dc)
     - Loading new contracts
     - Calling and encoding 
     - Screenshot 2024-04-17 at 22 13 57
     - Screenshot 2024-04-17 at 22 33 05
     - Sending
     - Deletion of contracts and adding of implementations

### CODE REVIEW

**Comment by 2xic** [comment]
* **#322** <a href="https://app.graphite.dev/github/pr/smlxl/evm.codes/322?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @2xic and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by mcmxcdev** [comment]
I just deleted one character from the contract address, leading to this overlap:

image

File: components/ContractViewer/ContractCodeEditor.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { useTheme } from 'next-themes'
 
 const ContractCodeEditor = ({ code, line, column, ...props }: any) => {
   const [codeEditor, setCodeEditor] = useState<any>(null)
-  const { theme } = useTheme()
+  const { resolvedTheme } = useTheme()
```
Comment (mcmxcdev) [inline components/ContractViewer/ContractCodeEditor.tsx:8]: I read into it and its weird that `theme` doesn't react to system preference, but whatevs. Thanks for the fix

File: components/ContractViewer/ContractTreeView.tsx [hunk 2]
```diff
@@ -1,8 +1,5 @@
-import { HorizontalRule } from '@mui/icons-material'
-import ChevronRightIcon from '@mui/icons-material/ChevronRight'
-import ExpandMoreIcon from '@mui/icons-material/ExpandMore'
 import { TreeItem } from '@mui/x-tree-view'
-import { TreeView } from '@mui/x-tree-view/TreeView'
+import { SimpleTreeView } from '@mui/x-tree-view/SimpleTreeView'
```
Comment (mcmxcdev) [inline components/ContractViewer/ContractTreeView.tsx:2]: Whats the difference between `TreeView` and `SimpleTreeView` ?

File: components/ContractViewer/DeploymentInfo.ts [hunk 3]
```diff
@@ -204,13 +223,15 @@ export const useDeployments = () => {
       address: string,
       context?: DeploymentInfo,
       loadImplementation = true,
+      invalidateRoute = true,
     ) => {
+      console.log('loading?')
```
Comment (mcmxcdev) [inline components/ContractViewer/DeploymentInfo.ts:228]: Remove this

File: components/ContractViewer/Header.tsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // TODO: make more generic components such as this
 const ContractViewerHeader = ({ children }: any) => {
   return (
-    <div className="w-full p-2 bg-gray-100 dark:bg-black-700 pr-6 h-16">
+    <div className="w-full p-2 bg-gray-100 dark:bg-black-700 h-[60px]">
```
Comment (mcmxcdev) [inline components/ContractViewer/Header.tsx:4]: Better to use paddings to achieve a desired height, but ok for evm.codes

File: components/ContractViewer/index.tsx [hunk 5]
```diff
@@ -89,23 +106,35 @@ const ContractViewerInner = () => {
...
     }
-    // NOTE: do not add dependencies here or it will cause an infinite loop (idk why)
+
+    void restoreSession()
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [router])
 
   // TODO: fix router to support contract implementations added by user
   // (currently only top-level contracts are supported)
-  useEffect(() => {
```
Comment (mcmxcdev) [inline components/ContractViewer/index.tsx:102]: Remove comment above this too which is now not connected to anything

File: components/ContractViewer/index.tsx [hunk 6]
```diff
@@ -114,13 +143,29 @@ const ContractViewerInner = () => {
...
+                    statusCallback: setStatus,
+                  })
+                }
+                disabled={status?.status === 'loading'}
+              />
+
+              {status?.status === 'loading' ? <CircularProgress /> : ''}
+            </div>
+            <div className="flex">
+              <p style={{ color: 'red' }}>
```
Comment (mcmxcdev) [inline components/ContractViewer/index.tsx:165]:
```suggestion
              <p class="text-red">
```

File: components/ui/vector/Remix.tsx [hunk 7]
```diff
@@ -74,6 +74,12 @@ function SvgRemix(props: any) {
         <symbol id="remix_svg__ri-search-line" viewBox="0 0 32 32">
           <path d="M24.041 22.156l5.711 5.709-1.887 1.887-5.709-5.711a11.871 11.871 0 01-7.472 2.625h-.018.001c-6.624 0-12-5.376-12-12s5.376-12 12-12 12 5.376 12 12v.017c0 2.838-.99 5.446-2.643 7.495l.018-.023zm-2.674-.989A9.27 9.27 0 0024 14.682v-.015.001a9.33 9.33 0 00-9.333-9.333 9.33 9.33 0 00-9.333 9.333 9.33 9.33 0 009.333 9.333h.015a9.273 9.273 0 006.487-2.635l-.002.002.2-.2z" />
         </symbol>
+        <symbol id="remix_svg__ri-close-large-line" viewBox="0 0 32 32">
+          <path d="M14.115 16 3.724 5.609 5.61 3.723l10.391 10.39 10.391-10.39 1.886 1.886L17.887 16l10.391 10.39-1.886 1.886-10.391-10.391L5.61 28.276 3.724 26.39z" />
+        </symbol>
+        <symbol id="remix_svg__ri-add-fill" viewBox="0 0 32 32">
+          <path d="M14.667 14.667v-8h2.667v8h8v2.667h-8v8h-2.667v-8h-8v-2.667h8z"></path>
+        </symbol>
```
Comment (mcmxcdev) [inline components/ui/vector/Remix.tsx:82]: What is this for?

File: components/ui/vector/icons.remixicon [hunk 8]
```diff
@@ -1 +1 @@
-search-line,contrast-2-fill,arrow-down-s-line,arrow-up-s-line,gas-station-fill,arrow-go-forward-line,arrow-left-s-line,arrow-right-s-line,checkbox-circle-line,command-line,contrast-2-line,error-warning-line,git-branch-line,github-fill,home-2-line,information-line,links-line,play-circle-line,question-line,reply-fill
+search-line,contrast-2-fill,arrow-down-s-line,arrow-up-s-line,gas-station-fill,arrow-go-forward-line,arrow-left-s-line,arrow-right-s-line,checkbox-circle-line,command-line,contrast-2-line,error-warning-line,git-branch-line,github-fill,home-2-line,information-line,links-line,play-circle-line,question-line,reply-fill,close-large-line,add-fill
```
Comment (mcmxcdev) [inline components/ui/vector/icons.remixicon:1]: Also don't know what this does

File: package.json [hunk 9]
```diff
@@ -27,10 +27,8 @@
     "@mdx-js/loader": "^2.3.0",
     "@mdx-js/react": "^2.3.0",
     "@monaco-editor/react": "^4.6.0",
-    "@mui/icons-material": "^5.15.2",
```
Comment (mcmxcdev) [inline package.json:30]: Thx for dependency cleanup!

**Comment by mcmxcdev** [comment]
Ideally, the MUI inputs can be replaced with already existing `Input` too at some point for more consistency.

File: components/ContractViewer/ContractCodeEditor.tsx (hunk 1 above)
Comment (2xic) [reply components/ContractViewer/ContractCodeEditor.tsx:8]: Yeah, I also found that a weird api design  🤷‍♂️

File: components/ContractViewer/ContractTreeView.tsx (hunk 2 above)
Comment (2xic) [reply components/ContractViewer/ContractTreeView.tsx:2]: It's the same! Or to be more precise `SimpleTreeView` is the new version of `TreeView`.

I had to [update the package](https://mui.com/x/migration/migration-tree-view-v6/#%E2%9C%85-use-simpletreeview-instead-of-treeview) to get text copying working

File: components/ContractViewer/index.tsx (hunk 5 above)
Comment (2xic) [reply components/ContractViewer/index.tsx:102]: Moved the comment to a better place as I still think it's nice to keep around

File: components/ui/vector/Remix.tsx (hunk 7 above)
Comment (2xic) [reply components/ui/vector/Remix.tsx:82]: The way we handle icons 😱

It's before my time, I think we should migrate to something else at some point ... I just followed the existing [instructions](https://github.com/smlxl/evm.codes/tree/8401a1383462e4873c99ea3ceabd09c3cf392005/components/ui/vector) for adding the new icons

File: components/ui/vector/icons.remixicon (hunk 8 above)
Comment (2xic) [reply components/ui/vector/icons.remixicon:1]: Same thing as [here](https://github.com/smlxl/evm.codes/pull/322#discussion_r1570396545)

**Comment by 2xic** [comment]
> Ideally, the MUI inputs can be replaced with already existing `Input` too at some point for more consistency.

💯 Yeah, I agree with this.  I wanted to get it done with this PR, but it would go past the allowed time for this cleanup :'(

**Review by mcmxcdev — APPROVED** [review]

**Pull request merged**